nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -127,28 +127,47 @@ BtNavigator::on_configure(const rclcpp_lifecycle::State & /*state*/)
   blackboard_->set<int>("number_recoveries", 0);  // NOLINT
 
   // Get the BT filename to use from the node parameter
-  std::string bt_xml_filename;
-  get_parameter("bt_xml_filename", bt_xml_filename);
+  get_parameter("default_bt_xml_filename", default_bt_xml_filename_);
+
+  if (!loadBehaviorTree(default_bt_xml_filename_)) {
+    RCLCPP_ERROR(get_logger(), "Error loading XML file: %s", default_bt_xml_filename_.c_str());
+    return nav2_util::CallbackReturn::FAILURE;
+  }
+
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+bool
+BtNavigator::loadBehaviorTree(const std::string & bt_xml_filename)
+{
+  // Use previous BT if it is the existing one
+  if (bt_xml_filename != "" && current_bt_xml_filename_ == bt_xml_filename) {
```

Comment:
What's with the empty string check? You're checking then that they're the same AND also empty, which is a special edge case, not the general checking if its the same since it wouldn't be empty string in any practical execution after loading.

Reply:
I think you mean to delete the empty string check.

Reply:
Deleted